pyqt-build

```diff
@@ -0,0 +1,26 @@
+#!/bin/sh
+
+PYTHON_VERSION=$(${PYTHON} -c 'import sys; print(sys.version[:3])')
+
+sip-install \
+    --verbose \
+    --build-dir=${FLATPAK_BUILDER_BUILDDIR}/tmp \
+    --target-dir=${FLATPAK_DEST}/lib/python${PYTHON_VERSION}/site-packages \
+    --enable=QtCore \
+    --enable=QtDBus \
+    --enable=QtGui \
+    --enable=QtMultimedia \
+    --enable=QtNetwork \
+    --enable=QtWidgets \
+    --confirm-license \
+    --qt-shared \
+    --no-designer-plugin \
+    --no-qml-plugin \
+    --no-dbus-python \
+    --no-tools \
+    --concatenate 1 \
+    --debug \
+    --no-docstrings \
+    --qmake-setting QMAKE_CFLAGS="$CFLAGS" \
+    --qmake-setting QMAKE_CXXFLAGS="$CXXFLAGS" \
+    --qmake-setting QMAKE_LFLAGS="$LDFLAGS"
```

Comment:
Could you reorder this file to make it as similar as possible to the old `pyqt-configure` script?

This would make it much easier to review the actual changes. :slightly_smiling_face:

Reply:
Is it good for you now ?

Reply:
Yes, this is much better, thank you. Thank you as well for splitting those other merge requests, this make sit all easier on my little brain. :slightly_smiling_face: 

I'm a bit disappointed that neither Git nor Github see that the new script is almost the same as the old one, and insists on treating them as a removal and an addition.

Nevertheless, now we can compare the two manually:

```diff
--- pyqt-configure	2020-02-10 15:02:33.256727180 +0100
+++ pyqt-build	2020-02-10 15:01:41.442990786 +0100
@@ -2,24 +2,25 @@
 
 PYTHON_VERSION=$(${PYTHON} -c 'import sys; print(sys.version[:3])')
 
-${PYTHON} configure.py \
-    --assume-shared \
-    --concatenate \
+sip-install \
+    --qt-shared \
+    --concatenate 1 \
     --confirm-license \
-    --destdir=${FLATPAK_DEST}/lib/python${PYTHON_VERSION}/site-packages \
+    --debug \
+    --verbose \
+    --build-dir=${FLATPAK_BUILDER_BUILDDIR}/tmp \
+    --target-dir=${FLATPAK_DEST}/lib/python${PYTHON_VERSION}/site-packages \
     --enable=QtCore \
+    --enable=QtDBus \
     --enable=QtGui \
     --enable=QtMultimedia \
     --enable=QtNetwork \
     --enable=QtWidgets \
+    --no-dbus-python \
     --no-designer-plugin \
-    --no-dist-info \
     --no-docstrings \
     --no-qml-plugin \
-    --no-qsci-api \
-    --no-sip-files \
-    --no-stubs \
     --no-tools \
-    QMAKE_CFLAGS_RELEASE="$CFLAGS" \
-    QMAKE_CXXFLAGS_RELEASE="$CXXFLAGS" \
-    QMAKE_LFLAGS_RELEASE="$LDFLAGS"
+    --qmake-setting QMAKE_CFLAGS="$CFLAGS" \
+    --qmake-setting QMAKE_CXXFLAGS="$CXXFLAGS" \
+    --qmake-setting QMAKE_LFLAGS="$LDFLAGS"
```

Below I'm going to try and review the above diff piece by piece, manually emulating the Github review UI.

```diff
-    --concatenate \
+    --concatenate 1 \
```

This surprised me, but looking at their old/new build systems, this change is entirely correct.

```diff
+    --debug \
+    --verbose \
```

This surprised me as well, because I thought `--debug` would enable verbose/debug output during the build, and I was wondering the difference with `--verbose`. It turns out the former is about generating builds with debuginfos (to be used by e.g `gdb`) so this is also entirely correct.

I wonder if our current builds just don't have debuginfos. :thinking:

```diff
+    --enable=QtDBus \
```

Why are you adding this? Is it something we're currently missing?

```diff
+    --no-dbus-python \
```

I'm guessing this is related to enabling `QtDBus`? This seems like [the right thing to do](https://dbus.freedesktop.org/doc/dbus-python/index.html#problems-and-alternatives).

```diff
-    --no-sip-files \
```

I seem to recall something about those "sip files" where not installing them would break something else's build later down, but I really can't remember what it was exactly. :disappointed:

Reply:
> I wonder if our current builds just don't have debuginfos. thinking

I sincerely have no idea.

> ```diff
> +    --enable=QtDBus \
> ```
> 
> Why are you adding this? Is it something we're currently missing?

Whoops. Leftover. It'll be required by the next Picard version.